code/modules/lighting/lighting_overlay.dm

```diff
@@ -127,8 +127,10 @@
 	return
 
 /atom/movable/lighting_overlay/Destroy()
+	// Do not remove this from SSlighting.overlay_queue
```

Comment:
What if lighting is suspended and it hard deletes after failing to GC?

Reply:
As per discord convo: SSlighting is never suspended in code, such an event would be rare/out-of-scope. The chance of the lighting overlay being hard deleted before SSlighting could process it is rare. Finally, the chance of a lighting_overlay being deleted which only occurs during a dynamic light change is also rare. As such it's rare^3 * out-of-scope and is fine.

Reply:
Actually even if it hard del the list reference would just be set to null, and SSlighting would be fine with that. QDELETED is a null check as well